soc: arm: stm32: Add support for STM32L152xC SoCs

noelleclement opened the pull request:
STM32L152xC SoC differs from other L1 SoCs in RAM (32KiB) and
flash (256KiB) size, EEPROM size (8Kib) and amount of interrupts 
(57, see STM32Cube). Devicetree and Kconfig support. 

Signed-off-by: Noelle Clement <noelleclement@hotmail.com>

gmarull commented on dts/arm/st/l1/stm32l152.dtsi:
```diff
@@ -1,11 +1,7 @@
- * the STM32L152 features an LCD controller.
- */
-#include <st/l1/stm32l151.dtsi>
+#include <st/l1/stm32l1.dtsi>
```
Can you explain a bit more this change?

noelleclement replied:
Of course, as I explained in #30790:

It is not needed for it to work, but this is much more in line with the abstraction 'levels' when looking at the other SoC models. If basing it on 151 is preferred, I can change it back, but I would say this is preferred since it might help overcome abstraction issues later on (if things need to be implemented that are specifically for 152, and not 151 etc)

gmarull replied:
So I think it is better to add a separate commit for this change, as it is not needed for the new additions.

noelleclement replied:
Done

erwango replied:
I'm a bit puzzled by this change.
In general we're trying to take advantage of SoC commonalities to avoid code duplication and related issues (maintenance, copy/paste issues, ..). So I don't think we should go this way.

noelleclement replied:
In the end, this is your (ST's) call I think, since you're more familiar with the common practices regarding hardware abstraction for the STM32 SoCs. 
Could you discuss this, and let me know if I need to change this back or not?

erwango replied:
> if things need to be implemented that are specifically for 152, and not 151 

If you compare 151 vs 152, IIRC the only diff is the LCD support. So at the end the L152 should only include L151 + LCD node. So I prefer to revert this and remain consistent across STM32 series.

noelleclement committed:
soc: arm: stm32: Add support for STM32L152xC SoCs

STM32L152xC SoC differs from other L1 SoCs in RAM (32KiB) and
flash (256KiB) size, EEPROM size (8Kib) and amount of interrupts
(57, see STM32Cube). Devicetree and Kconfig support.

Signed-off-by: Noelle Clement <noelleclement@hotmail.com>

noelleclement commented:
Removed the changes in dts/arm/st/l1/stm32l152xc.dtsi
